registry/docker/registry.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2017 tsuru authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package docker
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/tsuru/tsuru/registry"
+)
+
+func init() {
+	registry.Register("docker", &dockerRegistry{})
+}
+
+type dockerRegistry struct {
+	HTTPClient *http.Client
+	server     string
+}
+
+// RemoveAppImages removes an image manifest from a remote registry v2 server, returning an error
+// in case of failure.
+func (r *dockerRegistry) RemoveAppImages(appName string) error {
+	image := fmt.Sprintf("tsuru/app-%s", appName)
+	err := r.client()
+	if err != nil {
+		return err
+	}
+	tags, err := r.getImageTags(image)
+	if err != nil {
+		return err
+	}
+	for _, tag := range tags {
+		digest, err := r.getDigest(image, tag)
+		if err != nil {
+			fmt.Printf("failed to get digest for image %s/%s:%s on registry: %v\n", r.server, image, tag, err)
+			continue
+		}
+		err = r.removeImage(image, digest)
+		if err != nil {
+			if strings.Contains(err.Error(), "405") {
```

Comment:
It would be nice to set a specific error here (something like `ErrMethodNotAllowed`), instead of parsing the error message.